language-service/src/parser/jsonParser.ts

```diff
@@ -709,14 +714,53 @@ export class ObjectASTNode extends ASTNode {
 				seenKeys[key] = node.value;
 				unprocessedProperties.push(key);
 			}
-			
 		});
 
+		let findMatchingProperties = (propertyKey: string): ASTNodeMap => {
+			let result: ASTNodeMap = Object.create(null);
+			const compareKey: string = propertyKey.toUpperCase();
+
+			Object.keys(seenKeys).forEach((propertyName: string) => {
+				if (propertyName.toUpperCase() === compareKey) {
+					result[propertyName] = seenKeys[propertyName];
```

Comment:
I wonder if you can avoid all these functions by just putting properties with ignoreKeyCase=true into seenKeys in upper case from the beginning. And then just check `seenKeys[ignoreKeyCase ? propertyName.toUpperCase() : propertyName]`

Reply:
It can't be done easily because that would require matching the property against all possible property schemas (which is generally what this is trying to do anyway).  The implementation here starts from the schema which is assumed to use consistent casing for property definitions making the look-ups easier.